### ISSUE TITLE
Work without "official support"

### ISSUE THREAD

**Issue by badicsalex** [theHamsta/nvim-semantic-tokens#8]
Hi,

Since https://github.com/neovim/neovim/pull/15723 has completely stalled, would you be interested in including the actual LSP functionality into this plugin? I've been using that code for a year now by just copying it to my dotfiles, but your code looks way cleaner and I would rather use it as a plugin. 

I think I can create a PR too if you want.

**Comment by theHamsta** [comment]
Sounds like a great idea! At least we could have it as a fallback until the feature lands in Neovim

**Comment by theHamsta** [comment]
@badicsalex should now also work without official support.

**Comment by badicsalex** [comment]
@theHamsta, sounds great!
I planned to do something about this during the weekend... seems like that something is going to be just testing :)

**Comment by badicsalex** [comment]
I managed to make it work, but it took a bit of debugging. I ran into some problems:

1. Arch only has neovim-0.7.2-3, which apparently still uses the "old" `protocol.resolve_capabilities`, which did not work very well with your `vim.lsp._request_name_to_capability` monkey patch. I undo the patch from my local .lua for now.
2. With `rust_analyzer`, the main method of refreshing the highlights is to listen for the refresh event and calling `on_refresh`. This had to be manually set up. I think it would be reasonable for the plugin to do it during `setup()`, but there might be language servers or setups where this does not work all that well.
3. I found no way of fully rewriting the table highlighter other than directly setting the `token_map` of the module and then calling `reset()`. The fact that I need to manually call `reset()` was somewhat hard to figure out. Maybe I'm misunderstanding the whole "preset" concept, but it would be easier for me to provide a whole replacement map in the `setup(config)` parameter

I can live with all of these, and this is more of an FYI. Thanks for the work you put into the project.
